packages/core/src/components/axes/ruler.ts

```diff
@@ -0,0 +1,187 @@
+		));
+
+		/**
+		 * Find matches, reduce is used instead of filter
+		 * to only get elements which belong to the same axis coordinate
+		 */
+		const scaledValuesMatches: number[] = scaledData.reduce((acc, cur) => {
+			const sampleAccValue = acc[0];
+
+			// if current value is bigger than already existing values forget it
+			if (sampleAccValue && cur > sampleAccValue) {
+				return acc;
+			}
+
+			// there's a match and cur is either less then or equal to already stored values
+			if (pointIsMatch(cur, x)) {
+				if (sampleAccValue && cur < sampleAccValue) {
+					// there's a closer data point in the threshold area, so reinstantiate array
+					acc = [cur];
+				} else {
+					// cur is equal to already stored values, there's another match on the same coordinate
+					acc.push(cur);
+				}
+			}
+
+			return acc;
+		}, []);
+
+		// some data point match
+		if (scaledValuesMatches.length > 0) {
+			const sampleMatch = scaledValuesMatches[0];
+
+			const highlightItems = this.services.cartesianScales.getDataFromDomain(
+				mainXScale.invert(sampleMatch)
+			).filter(d => d.value);
+
+			const hoveredElements = dataPoints.filter((d, i) =>
+				scaledValuesMatches.includes(
+					Number(this.services.cartesianScales.getDomainValue(d))
+				)
+			);
+
+			/** if we pass from a trigger area to another one
+			 * mouseout on previous elements won't get dispatched
+			 * so we need to do it manually
+			 */
+			if (
+				this.hoveredElements &&
+				this.hoveredElements.size() > 0 &&
+				!isEqual(this.hoveredElements, hoveredElements)
+			) {
+				this.hoveredElements.dispatch("mouseout");
+			}
+
+			hoveredElements.dispatch("mouseover");
+
+			// set current hovered elements
+			this.hoveredElements = hoveredElements;
+
+			this.services.events.dispatchEvent("show-tooltip", {
+				hoveredElement: line,
+				multidata: highlightItems,
+				type: TooltipTypes.GRIDLINE
+			});
+
+			ruler.attr("opacity", 1);
+
+			// line snaps to matching point
+			line.attr("y1", yScaleStart)
+				.attr("y2", yScaleEnd)
+				.attr("x1", sampleMatch)
+				.attr("x2", sampleMatch);
+		} else {
+			ruler.attr("opacity", 0);
+			dataPoints.dispatch("mouseout");
+		}
+	}
+
+	hideRuler() {
+		const svg = this.parent;
+		const ruler = DOMUtils.appendOrSelect(svg, "g.ruler");
+		const dataPoints = svg.selectAll("[role=graphics-symbol]");
+
+		dataPoints.dispatch("mouseout");
+		ruler.attr("opacity", 0);
+	}
+
+	/**
+	 * Adds the listener on the X grid to trigger multiple point tooltips along the x axis.
+	 */
+	addBackdropEventListeners() {
+		const self = this;
+
+		this.backdrop
+			.on("mousemove mouseover", function() {
+				const chartContainer = self.services.domUtils.getMainSVG();
+				const pos = mouse(chartContainer);
+
+				self.showRuler(pos);
+			})
+			.on("mouseout", function() {
+				self.hideRuler();
+				self.services.events.dispatchEvent("hide-tooltip", {});
+			});
+	}
+
+	drawBackdrop() {
```

Comment:
It is needed to attach mouse listeners

Reply:
that means we can get rid of the `grid.ts` event listeners for tooltip...?

Reply:
we discussed this and while we don't think it would be good to remove listeners from `grid.ts` as it would affect other charts too (where ruler does not exist), what we can do is add the possibility to pass a boolean option to the `Grid` class constructor to disable the event listeners for specific cases (right now only when ruler is active, which means only on the linechart)

Reply:
just to recap after latest discussion, I'm now going to disable mouse listeners on grid and enable ruler interaction for every other chart and scale type 😅